test/handlebars.js

```diff
@@ -23,6 +23,8 @@ var tmpl = [
   'ifConditional1: {{#if (equals "foo" foo)}}{{upper name}}{{/if}}',
   'ifConditional2: {{#if (equals "baz" bar)}}{{upper name}}{{/if}}',
   'ifConditional3: {{#if foo}}{{upper name}}{{/if}}',
+  'ifConditional4: {{#if false}}{{upper name}}{{/if}}',
```

Comment:
I don't get why when `equals` returns boolean `false` it does not work when this passes.

Reply:
In the comment above, I explain how the async helper `equals` will actually return a string (the async id) during the rendering pass, then the id is resolved. Since `false` here is a literal value that `handlebars` handles, this works fine.